components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -8202,9 +8202,16 @@ public static String getAnalyticsServerPassword() {
      * @param accessExp        - Value of the ACCESSED Expiry Type
      * @return - The cache object
      */
-    public static Cache getCache(final String cacheManagerName, final String cacheName, final long modifiedExp,
+    public synchronized static Cache getCache(final String cacheManagerName, final String cacheName, final long modifiedExp,
```

Comment:
Do we need to synchronize this at the method level? If this method is abundantly used, then having the method synchronized would produce unnecessary contentions. If that's the case can we only wrap the object creation with a sync block?

Reply:
When starting up the server and caches are getting initialised and intermittently some threads try to access the caches simultaneously, so when two thread already inside this method and  trying to create the same cache if not present, then following kind of sample error occurs.
`javax.cache.CacheException: Cache $__local__$.gatewayUsernameCache already exists`
As per my understanding only in startup time this method get's invoked which creates the relevant caches.